clang/utils/TableGen/ClangProgModelBuiltinEmitter.cpp

```diff
@@ -848,6 +916,226 @@ void BuiltinNameEmitter::EmitQualTypeFinder() {
   OS << "\n} // Bultin2Qual\n";
 }
 
+JSONBuiltinInterfaceEmitter::TypeDesc::TypeDesc(const Record *T)
+    : Name(T->getValueAsString("Name")),
+      VecWidth(T->getValueAsInt("VecWidth")),
+      ElementSize(T->getValueAsInt("ElementSize")),
+      IsPointer(T->getValueAsBit("IsPointer")),
+      IsConst(T->getValueAsBit("IsConst")),
+      IsVolatile(T->getValueAsBit("IsVolatile")) {
+  if (T->getValueAsBit("IsInteger")) {
+    if (ElementSize == 1)
+      ElementType = BOOL;
+    else
+      ElementType = T->getValueAsBit("IsSigned") ? INT : UINT;
+  }
+  if (T->getValueAsBit("IsFloat"))
```

Comment:
I wonder if this needs an assert, can the element-type be both an integer and a float?  What is the expected behavior there?

Reply:
They are mutually exclusive, I added an assert.